test/views/SubmissionsViewSpec.scala

```diff
@@ -0,0 +1,93 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views
+
+import java.time.format.DateTimeFormatter
+import java.time.{Instant, LocalDate, ZoneId, ZonedDateTime}
+
+import models.UcrBlock
+import models.notifications.NotificationFrontendModel
+import models.submissions.SubmissionFrontendModel
+import play.api.mvc.{AnyContent, Request}
+import play.api.test.FakeRequest
+import play.twirl.api.Html
+import testdata.CommonTestData._
+import testdata.MovementsTestData.exampleSubmissionFrontendModel
+import testdata.NotificationTestData.exampleNotificationFrontendModel
+import views.html.view_submissions
+
+class SubmissionsViewSpec extends ViewSpec {
+
+  private implicit val request: Request[AnyContent] = FakeRequest().withCSRFToken
+
+  private def createView(submissions: Seq[(SubmissionFrontendModel, Seq[NotificationFrontendModel])] = Seq.empty): Html =
+    new view_submissions(main_template)(submissions)
+
+  "Submissions page" should {
+
+    "contain title" in {
+
+      createView().getElementById("title") must containText(messages("submissions.title"))
+    }
+
+    "contain correct table headers" in {
+
+      val page = createView()
+
+      page.getElementById("ucr") must containText(messages("submissions.ucr"))
+      page.getElementById("ucrType") must containText(messages("submissions.submissionType"))
+      page.getElementById("submissionAction") must containText(messages("submissions.submissionAction"))
+      page.getElementById("dateOfRequest") must containText(messages("submissions.dateOfRequest"))
+    }
+
+    "contain correct submission data" in {
+
+      val dateTime: Instant = ZonedDateTime
+        .of(LocalDate.parse("2019-10-31", DateTimeFormatter.ofPattern("yyyy-MM-dd")).atStartOfDay(), ZoneId.systemDefault())
+        .toInstant
+      val submission = exampleSubmissionFrontendModel(requestTimestamp = dateTime)
+      val notifications = Seq(exampleNotificationFrontendModel(timestampReceived = dateTime.plusSeconds(3)))
+
+      val page = createView(Seq((submission, notifications)))
+
+      page.getElementById(s"ucr-$conversationId").text() mustBe correctUcr
+      page.getElementById(s"ucrType-$conversationId").text() mustBe "DUCR"
+      page.getElementById(s"dateOfRequest-$conversationId").text() mustBe "31 Oct 2019 at 00:00"
+      page.getElementById(s"submissionAction-$conversationId").text() mustBe messages(s"submissions.arrival")
+    }
+
+    "contain MUCR and DUCR if Submission contains both" in {
+
+      val dateTime: Instant = ZonedDateTime
+        .of(LocalDate.parse("2019-10-31", DateTimeFormatter.ofPattern("yyyy-MM-dd")).atStartOfDay(), ZoneId.systemDefault())
+        .toInstant
+      val submission = exampleSubmissionFrontendModel(requestTimestamp = dateTime)
+        .copy(ucrBlocks = Seq(UcrBlock(ucr = correctUcr, ucrType = "M"), UcrBlock(ucr = correctUcr_2, ucrType = "D")))
```

Comment:
Cant we pass in  `ucrBlocks` to the method rather than calling `copy`?

Reply:
No, because this method accepts only ucr and ucrType. This is quite old implementation, before we were aware that there can be more UCRs. However this is a good call to change it now for the new service before it grew too big.